src/ducks/apps/index.js

```diff
@@ -409,28 +406,21 @@ export function fetchRegistryApps(lang, channel = DEFAULT_CHANNEL) {
   return dispatch => {
     dispatch({ type: FETCH_APPS })
     return cozy.client
-      .fetchJSON('GET', '/registry')
+      .fetchJSON('GET', `/registry?limit=150&channelLatestVersion=${channel}`)
       .then(response => {
         const apps = response.data
           .filter(app => !config.notDisplayedApps.includes(app.name))
           .filter(app => app.versions.dev && app.versions.dev.length) // only apps with versions available
         return Promise.all(
           apps.map(app => {
+            if (!app.latest_version) return false // skip
             return getFormattedRegistryApp(app, channel).catch(err => {
-              if (err.status === 404) {
-                console.warn(
-                  `No ${channel} version found for ${
-                    app.slug
-                  } from the registry.`
-                )
-              } else {
-                console.warn(
-                  `Something went wrong when trying to fetch more informations about ${
-                    app.slug
-                  } from the registry on ${channel} channel. ${err}`
-                )
-              }
-              return false // useful to skip in an array.map function
+              console.warn(
+                `Something went wrong when trying to fetch more informations about ${
+                  app.slug
```

Comment:
I suggest to use app.name instead for an improved readability.

Reply:
I would prefer the slug here, it will be better for debugging, won't be? The user can directly fire this message as issue for us.

Reply:
Ok for me.